lib/rucio/core/rule.py

```diff
@@ -2808,12 +3322,16 @@ def __evaluate_did_attach(eval_did: models.DataIdentifier, *, session: "Session"
 
         # Get immediate new child DID's
         with METRICS.timer('evaluate_did_attach.list_new_child_dids'):
-            new_child_dids = session.query(models.DataIdentifierAssociation)\
-                .with_hint(models.DataIdentifierAssociation, "INDEX_RS_ASC(contents contents_pk)", 'oracle')\
-                .filter(models.DataIdentifierAssociation.scope == eval_did.scope,
-                        models.DataIdentifierAssociation.name == eval_did.name,
-                        models.DataIdentifierAssociation.rule_evaluation == True).all()  # noqa
-
+            stmt = select(
+                models.DataIdentifierAssociation
+            ).with_hint(
+                models.DataIdentifierAssociation, 'INDEX_RS_ASC(CONTENTS CONTENTS_PK)', 'oracle'
+            ).where(
+                and_(models.DataIdentifierAssociation.scope == eval_did.scope,
+                     models.DataIdentifierAssociation.name == eval_did.name,
+                     models.DataIdentifierAssociation.rule_evaluation == true())  # pylint: disable=C0121:singleton-comparison  # noqa: E712
```

Comment:
I’m surprised, why are these directives to checkers necessary? I though that was the point of using `true()`.

Reply:
They are not, I likely forgot to remove them when I rewrote this to use `true()` previously. Removed them now